go/libraries/doltcore/doltdb/table.go

```diff
@@ -34,43 +32,30 @@ import (
 	"github.com/dolthub/dolt/go/store/types"
 )
 
-const (
```

Comment:
These regex were following MySQL restrictions for unquoted identifiers, but any questionable unquoted identifers shouldn't make it this far. 
For example
```sql
create table 123(i int); -- syntax error in vitess
create table a-a(i int); -- syntax error in vitess
create table []<>{}(i int); -- syntax error in vitess
```